### PR TITLE
Added recursive test resolution capability

### PR DESCRIPTION
**Pull request by kunagpal**
* Re-organized integration test collections into respective individual folders
* Added recursive directory traversal in tests using `recursive-readdir`

### CODE REVIEW

**Comment by shamasis** [comment]
Do we need to put all files in individual folders? I was thinking of putting all old integration tests into a `newman-v2-legacy` folder and rest stays in root and more than one related tests are put in folder.

**Comment by kunagpal** [comment]
That tatters things a bit. Given the current structure, it makes more sens to put tests in their own folders, or none at all. Also, we haven't yet decided on what constitutes an old integration test.

File: npm/test-integration.js [hunk 1]
```diff
@@ -51,7 +59,7 @@ module.exports = function (exit) {
 
             // run tests using the consolidated test set as a guide
             async.map(suites, function (test, next) {
-                console.info(` - ${test.name}`);
+                console.info(` - ${test.name.split('/')[2]}`);
```
Comment (shamasis) [inline npm/test-integration.js]: what's this @kunagpal ?

Comment (shamasis) [inline npm/test-integration.js]: what if I do not have a folder????

Comment (kunagpal) [inline npm/test-integration.js]: Interesting point. Tests have been arranged within folders, but that shouldn't be taken as a guarantee. I'll improve the logic here in the next commit.

**Comment by kunagpal** [comment]
@shamasis That has already been handled, integration tests have been moved to folders only for those with multiple files under them.
